src/astro/files/base.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import create_file_location
+from astro.files.type import create_type_factory
+
+
+class File:
+    def __init__(self, path: str, conn_id: Optional[str] = None):
```

Comment:
Could we have an optional parameter: `type`, in case the user does not Astro to guess the filetype by the file extension.

Reply:
Done.